Allow targeting specific battery, inverter and EV charger types

llucax opened the pull request:
This is useful to dispatch to only solar inverters, or hybrid EV chargers for example.

llucax committed:
Clear release notes

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Add a way to target specific battery, inverter and EV charger types

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax commented:
We decided to go with this approach because we want maximum flexibility for targeting components. We are not allowing to mix IDs with categories/types because it could be hard to *deduplicate*, if a passed ID is contained in a passed category, we need to check one by one.

Personally I'm not completely convinced about this argument, but I guess it is part of a separate discussion, and if we ever decide to allow mixing anything, this PR is the right approach anyways.

~This also allows us to reuse [the common `ComponentCategoryMetadataVariant`](https://github.com/frequenz-floss/frequenz-api-common/blob/8b3342b05869bd9be608a5daf8af9cb0fac8c2c5/proto/frequenz/api/common/v1/microgrid/components/components.proto#L88-L99), so it should be easier to maintain and keep in sync.~

Marenz commented on proto/frequenz/api/dispatch/v1/dispatch.proto:
```diff
@@ -326,19 +336,42 @@ message TargetComponents {
+    repeated CategoryAndType categories = 1;
+  }
+
+  // A tuple of a (required) category and an (optional) type
```
I would even remove the `()` here

Marenz reviewed (commented):
LGTM, optional comment. No idea why protolint acts up.

llucax committed:
Use `github-check` reporter for protolint

Using `github-pr-review` requires elevated write permissions to the
repository, just to write a review, which can compromise the security
of the repository, by using the `pull_request_target` event which gives
the GitHub token write access, or using a custom token which complicates
the setup.

To simplify things we switch back to `github-check` reporter, which is
the repo-config default and doesn't need any write permissions, and
failed checks will still be shown in the diff view.

We also pin the protolint version to `0.52.0` to avoid any surprises
with future versions, also as repo-config does.

https://github.com/reviewdog/reviewdog?tab=readme-ov-file#reporter-github-checks--reportergithub-check

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax commented:
> LGTM, optional comment. No idea why protolint acts up.

imports not properly sorted, it seems it had some permission issues to report the failed checks more visibly in the code, I added a fix for that, we'll have to propagate to other projects via repo-config (I already updated my local repo-config so I don't forget).

llucax commented:
Actually, in repo-config we are using `github-check`, not sure why the reported was changed in this commit: 9b6e56995a64fbfb20e036e546256aee52074841. So I will revert to match what's in repo-config.

llucax commented:
Done.
